chia/types/spend_bundle.py

```diff
@@ -77,6 +80,27 @@ def not_ephemeral_additions(self):
 
         return result
 
+    def get_memos(self) -> Dict[bytes32, List[bytes]]:
+        """
+        Retrieves the memos for additions in this spend_bundle, which are formatted as a list in the 3rd parameter of
+        CREATE_COIN. If there are no memos, the addition coin_id is not included. If they are not formatted as a list
+        of bytes, they are not included. This is expensive to call, it should not be used in full node code.
+        """
+        memos: Dict[bytes32, List[bytes]] = {}
+        for coin_spend in self.coin_spends:
+            result = Program.from_bytes(bytes(coin_spend.puzzle_reveal)).run(
+                Program.from_bytes(bytes(coin_spend.solution))
+            )
+            for condition in result.as_python():
+                if condition[0] == ConditionOpcode.CREATE_COIN and len(condition) >= 4:
+                    # If only 3 elements (opcode + 2 args), there is no memo, this is ph, amount
+                    coin_added = Coin(coin_spend.coin.name(), bytes32(condition[1]), int_from_bytes(condition[2]))
+                    if type(condition[3]) != list:
```

Comment:
This is `get_memos` so it wants everything including the hints AND the additional memos.  That's why we use vanilla `run` rather than the rust `run_with_generator2`.  It's used all over the place but probably most extensively tested in test/wallet/rpc/test_wallet_rpc.py.

Reply:
ah, I wasn't paying attention to that. that makes sense then

Reply:
it makes me a bit nervous to have a dangerous function, like this, mixed in as just another member of `SpendBundle`. It
s critical this is never used by the full node and that it's only ever run one one's own spend bundles, that can be trusted. But that's something to address later.

Reply:
I'm unresolving as the last concerns seems like it should result in some action.  I've been told that all of the concerns raised here are applicable to main so we should keep this list complete for creating other tasks to fix these things on main.

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/9907